.circleci/config.yml

```diff
@@ -59,6 +59,66 @@ jobs:
           name: Post Test Results to GitHub
           command: bundle exec danger || echo "danger failed"
           when: always # Run this even when tests fail
+
+  "Execute tests on macOS (Xcode 11.0.0, Ruby 2.5)":
+    macos:
+      xcode: "11.0.0"
+    environment:
+      CIRCLE_TEST_REPORTS: ~/test-reports
+      LC_ALL: en_US.UTF-8
+      LANG: en_US.UTF-8
+    shell: /bin/bash --login -eo pipefail
+    steps:
+      - checkout
+
+      - restore_cache:
+          key: v1-macos-gems-{{ checksum "Gemfile" }}-{{ checksum "fastlane.gemspec" }}
+      - restore_cache:
+          keys:
+            - v3-homebrew-{{ epoch }}
+            - v3-homebrew
+      - run:
+          name: Setup Build
+          command: |
+            mkdir -p ~/test-reports
+            echo "2.5" > .ruby-version
+            gem update --system
+            brew install shellcheck
+            bundle check --path .bundle || bundle install --jobs=4 --retry=3 --path .bundle
+
+      - save_cache:
+          key: v3-homebrew-{{ epoch }}
+          paths:
+            - /usr/local/Homebrew
+      - save_cache:
+          key: v1-macos-gems-{{ checksum "Gemfile" }}-{{ checksum "fastlane.gemspec" }}
+          paths:
+            - .bundle
+
+      - run:
+          name: Check PR Metadata
+          command: bundle exec danger || echo "danger failed"
+
+      - restore_cache:
+          key: v1-{{ arch }}-rubocop
+
+      - run: bundle exec fastlane execute_tests
+
+      - save_cache:
+          key: v1-{{ arch }}-rubocop
+          paths:
+            - ~/.cache/rubocop_cache
+      - store_test_results:
+          path: ~/test-reports
+      - store_artifacts:
+          path: ~/test-reports/rspec
+          destination: test-reports
+
+      - run:
+          name: Post Test Results to GitHub
+          command: bundle exec danger || echo "danger failed"
+          when: always # Run this even when tests fail
```

Comment:
The only difference between `Execute tests on macOS (Xcode 9.0.1, Ruby 2.3)` and `Execute tests on macOS (Xcode 11.0.0, Ruby 2.5)` jobs is the `macos` parameter. It may be a bit hard to maintain these jobs, because we will need to make the changes in both (copy/paste most probably). Is it a way on circle CI to make some parts of the job reusable? Maybe [Reusable Commands](https://circleci.com/docs/2.0/reusing-config/#invoking-reusable-commands) could help here? 🤔

Reply:
The Ruby version is also different - but that's it. If there is a way to simplify the job definition, go for it. There probably is some way to extract jobs to files maybe?

Reply:
> If there is a way to simplify the job definition, go for it.

Maybe we could leverage the [YAML's anchors](https://docs.gitlab.com/ee/ci/yaml/README.html#anchors) feature:

> YAML has a handy feature called ‘anchors’, which lets you easily duplicate content across your document. Anchors can be used to duplicate/inherit properties, ....

Reply:
If Circle supports this, indeed this could work. I did so for Apache Cordova taht uses Travis: https://github.com/apache/cordova-paramedic/blob/master/.travis.yml The syntax is a bit ... unintuitive though. Best check the Circle CI docs again if they have something about structuring your jobs etc.